app/js/details.js

```diff
@@ -35,6 +35,14 @@
             // Will use the Cordova InAppBrowser plugin when deployed to a device. Opens a new window in
             // the simulator
             window.open(currentBook.amazon_url, '_blank', 'location=yes');
+        },
+        setIsFavorite: function () {
+            Books.data.fetch(function () {
+                var dataItem = currentBook;
+                if(dataItem.is_favorite==true)dataItem.set("is_favorite", false);
```

Comment:
Please do leave more whitespace and add blocks, something like this:
```javascript
if (dataItem.is_favorite==true) {
    dataItem.set("is_favorite", false);
} else {
    dataItem.set("is_favorite", true);
}
```
Alternatively you can write the whole thing like this:
```javascript
dataItem.set("is_favorite", !dataItem.is_favorite);
```